do-catalog/src/components/datasetDetail/DatasetHeader.vue

```diff
@@ -76,22 +139,46 @@ export default {
         this.dataset.data_source_name
       );
     },
-    showModal() {
+    showModal(mode) {
+      this.modalMode = mode;
       this.modalOpen = true;
     },
     hideModal() {
+      this.modalMode = null;
       this.modalOpen = false;
+    },
+    fetchSubscriptionInfo() {
+      if (this.dataset && this.dataset.id && !this.isPublicWebsite) {
+        this.$store.dispatch('doCatalog/fetchSubscriptionInfo', {
+          id: this.dataset.id,
+          type: this.isGeography ? 'geography' : 'dataset',
+          api_key: this.$store.state.user.api_key
```

Comment:
You can access to `User` state from the action, using `context.rootState`, no need to send it from here.

Reply:
changed